File: cv-creator-backend/src/main/java/at/friedrichbachinger/mainappfcb/rest/response/PublicPageResponse.java
package at.friedrichbachinger.mainappfcb.rest.response;

import java.util.Date;
import java.util.Set;

import at.friedrichbachinger.mainappfcb.entity.ImageDTO;
import at.friedrichbachinger.mainappfcb.entity.elements.Address;
import at.friedrichbachinger.mainappfcb.entity.elements.Experience;
import at.friedrichbachinger.mainappfcb.entity.elements.Hobby;
import at.friedrichbachinger.mainappfcb.entity.elements.Knowledge;
import at.friedrichbachinger.mainappfcb.entity.elements.Progression;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
public class PublicPageResponse {
	private Address address;
	private Set<Progression> progressions;
	private Set<Experience> experiences;
	private Set<Knowledge> knowledges;
	private Set<Hobby> hobbies;
	private Set<ImageDTO> images;
	private Date lastModifiedDate;
}

File: cv-creator-frontend/src/app/shared/models/public-page.response.ts
import { ExperienceEntity } from 'src/app/store/experience/experience.entity';
import { HobbyEntity } from 'src/app/store/hobby/hobby.entity';
import { AddressEntity } from 'src/app/store/address/address.entity';
import { ProgressionEntity } from 'src/app/store/progression/progression.entity';
import { KnowledgeEntity } from 'src/app/store/knowledge/knowlege.entity';
import { ImageEntity } from 'src/app/store/image/image.entity';

export interface IPublicPageResponse {
    address: AddressEntity;
    progressions: ProgressionEntity[];
    experiences: ExperienceEntity[];
    knowledges: KnowledgeEntity[];
    hobbies: HobbyEntity[];
    images: ImageEntity[];
    lastModifiedDate: Date;
}
